File: python/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        output = [i + 1 for i in range(len(nums))]
        for i in nums:
            output[i - 1] = 0
        end_of_list = 0
        for i in range(len(output)):
            if output[i] != 0:
                output[end_of_list], output[i] = output[i], output[end_of_list]
                end_of_list += 1
        while output and output[-1] == 0:
            output.pop()
        return output

File: rust/find-all-numbers-disappeared-in-an-array.rs
se std::collections::HashSet;

impl Solution {
    pub fn find_disappeared_numbers(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len() as i32 + 1;
        let nums_set: HashSet<i32> = nums.into_iter().collect();
        let mut output = Vec::new();
        for i in 1..n {
            if !nums_set.contains(&i) {
                output.push(i);
            }
        }
        return output;
    }
}
